File: Ejercicios Propuestos 1/Ejercicio 5/EJERCICIO_5.cpp
#include <iostream>

using namespace std;

int main() {
    int anio;
    int total_leidos = 0;
    int bisiestos_encontrados = 0;

    while (total_leidos < 10 && bisiestos_encontrados < 3) {
        cout << "Ingrese un año: ";
        cin >> anio;

        total_leidos++;

        if ((anio % 4 == 0 && anio % 100 != 0) || (anio % 400 == 0)) {
            cout << anio << " es un año BISIESTO." << endl;
            bisiestos_encontrados++;
        } else {
            cout << anio << " NO es un año bisiesto." << endl;
        }
        cout << "--- continuemos ----" << endl;
    }

    cout << "--- Proceso terminado ---" << endl;
    cout << "Total de años revisados: " << total_leidos << endl;
    cout << "Total de bisiestos encontrados: " << bisiestos_encontrados << endl;

    return 0;
}

File: Ejercicios Propuestos 1/Ejercicio 5/Ejercicio5.java
import java.util.Scanner;

public class Ejercicio5 {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            int totalLeidos = 0;
            int bisiestosEncontrados = 0;
            System.out.println("--- Analizador de Años Bisiestos ---");
            System.out.println("Condiciones de parada: 10 años leídos o 3 bisiestos hallados.");
            while (totalLeidos < 10 && bisiestosEncontrados < 3) {
                System.out.print("Introduzca un año: ");
                if (sc.hasNextInt()) {
                    int anio = sc.nextInt();
                    totalLeidos++;
                    if ((anio % 4 == 0 && anio % 100 != 0) || (anio % 400 == 0)) {
                        bisiestosEncontrados++;
                        System.out.println("-> " + anio + " ES bisiesto.");
                    } else {
                        System.out.println("-> " + anio + " NO es bisiesto.");
                    }
                } else {
                    System.out.println("Error: Por favor, introduzca un número válido.");
                    sc.next(); // Limpiar el buffer si el usuario escribe texto
                }
            }
            System.out.println("\n--- Resumen Final ---");
            System.out.println("Años procesados: " + totalLeidos);
            System.out.println("Bisiestos hallados: " + bisiestosEncontrados);
        } 
    }
}